### PR TITLE
Zero-Downtime with VersionedFields gem

### PR DESCRIPTION
**Pull request by nattfodd**
No breaking changes. It just adds module, which can be used with [versioned_fields](https://github.com/riboseinc/versioned_fields) gem to achieve zero-downtime migration.

Usage example can be found here: https://github.com/nattfodd/transcryptor_sample_app/pull/3/files

### CODE REVIEW

**Comment by ronaldtse** [comment]
@ribose-jeffreylau can you help check? Thanks!

File: lib/transcryptor/decoder.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      # Encoding will be performed by :attr_encypted itself
+      return public_send(migration_data.field) if @transcryptor_decode_migrated
+
+      field     = migration_data.field
+      current_v = migration_data.version
+
+      generate_versioned_fields!(field, current_v, options)
+
+      opts = encrypted_attributes[field]
+      public_send("#{field}_#{current_v}")
```
Comment (ribose-jeffreylau) [inline lib/transcryptor/decoder.rb]: Is this extra `public_send` necessary?

Comment (nattfodd) [reply lib/transcryptor/decoder.rb]: No, I just forgot to remove this line. Updated.

File: README.adoc [hunk 2]
```diff
@@ -215,6 +215,36 @@ Default options for old and new configuration are absolutelly the same as it is
 }
 ----
 
+== Zero downtime migration using VersionedFields (from version 0.3.0)
+
+You can use transcryptor to migrate your encrypted fields in zero downtime
+using https://github.com/riboseinc/versioned_fields[versioned_fields] gem.
+
+Just include `Transcryptor::Decoder` module and use `decode_with_previous_settings`
+to decode your fields with a specific settings:
```
Comment (ribose-jeffreylau) [inline README.adoc]:
```diff
-a specific settings
+specific settings
```

**Review by ribose-jeffreylau — CHANGES_REQUESTED** [review]

**Review by ribose-jeffreylau — APPROVED** [review]

**Pull request merged**